ccmlib/common.py

```diff
@@ -389,23 +389,46 @@ def make_cassandra_env(install_dir, node_path, update_conf=True, hardcode_java_v
 
     if hardcode_java_version:
         known_jvm_names = {
-            '8': ('8-openjdk', '8-jdk', '1.8.0-openjdk'),
-            '11': ('11-openjdk', '11-jdk', '1.11.0-openjdk'),
+            '8': ['1.8', '8'],
+            '11': ['11']
         }
-        assert hardcode_java_version in known_jvm_names.keys(), \
+        jvm_root_path = "/usr/lib/jvm/"
+        assert hardcode_java_version in known_jvm_names, \
             f"hardcode_java_version={hardcode_java_version} not supported in:\n{known_jvm_names}"
 
-        for java_path in pathlib.Path('/usr/lib/jvm/').iterdir():
-            if (java_path.is_dir
-                    and any(name in java_path.name for name in known_jvm_names[hardcode_java_version])
-                    and list(java_path.rglob('java'))):
-                env['JAVA_HOME'] = str(java_path)
-                break
-        else:
-            raise ArgumentError(f"java-8 wasn't found in /usr/lib/jvm/\n {list(pathlib.Path('/usr/lib/jvm/').iterdir())}")
+        java_home_path = get_java_home_path(pathlib.Path(jvm_root_path), known_jvm_names[hardcode_java_version])
+        if java_home_path is None:
+            raise ArgumentError(f"java-{hardcode_java_version} wasn't found in {jvm_root_path}")
 
+        env['JAVA_HOME'] = str(java_home_path.as_posix())
     return env
 
+def get_java_home_path(parent_path: pathlib.Path, hardcode_java_version: List[str]) -> [pathlib.Path]:
+    for java in pathlib.Path(parent_path).rglob('*/bin/java'):
+        if not java.is_file():
+            continue
+        if get_jvm_spec_version(java) in hardcode_java_version:
+            return java.parent.parent
+    return None
+
+java_version_property_regexp = re.compile(r'\s*(?:java\.specification\.version|java\.vm\.specification\.version)\s*=\s*([0-9.]+)\s*')
+
+def get_jvm_spec_version(java_bin_path: pathlib.Path) -> Optional[str]:
+    if not java_bin_path.is_file():
+        return None
+    try:
+        properties = subprocess.check_output([java_bin_path.as_posix(), '-XshowSettings:properties', '-version'], stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        return None
+
+    match = java_version_property_regexp.search(properties.decode('utf-8'))
+    if match is None:
+        return None
+    for version in match.groups():
+        return version
+    return None
+
+print(get_java_home_path(pathlib.Path("/usr/lib/jvm"), ["8", "1.8"]))
```

Comment:
please that this test code out. (or put it in in tests folder)

Reply:
done.

Reply:
go right ahead and add test code.

but this part if used in 5 cases where we need to run cassandra.

and we don't use java in master anymore, i.e. we won't run jmx or the old nodetool

I'm not sure in what context you are using this part, and chasing it the issue in it ? 
in using cassandra for the driver tests ?

Reply:
> go right ahead and add test code.
> 
In progress.
> but this part if used in 5 cases where we need to run cassandra.
> 
> and we don't use java in master anymore, i.e. we won't run jmx or the old nodetool
> 
> I'm not sure in what context you are using this part, and chasing it the issue in it ? in using cassandra for the driver tests ?

while runing `java-driver` and `cpp-rust-driver` tests localy on my PC I have faced this issue with ccm picking wrong java, as result whole suit either fails or getting stuck.

Reply:
running it with which scylla version ?

Reply:
sounds like you are running cassandra and not scylla, cause running scylla you should be getting into that code path....

Reply:
> go right ahead and add test code.

Done

Reply:
> sounds like you are running cassandra and not scylla, cause running scylla you should be getting into that code path....

Correct, running on cassandra.